WcaOnRails/app/models/competition.rb

```diff
@@ -755,8 +763,14 @@ def events_to_associated_events(events)
     end
   end
 
-  def self.search(query, params: {})
-    competitions = Competition.where(showAtAll: true)
+  def self.search(query, params: {}, current_resource_owner: nil)
+    if params[:managed_by_me].present?
+      raise WcaExceptions::BadApiParameter.new("Must be logged in to specify `managed_by_me`") unless current_resource_owner
+      raise WcaExceptions::BadApiParameter.new("Missing required scope `manage_competitions`") unless current_resource_owner.doorkeeper_token.scopes.include?("manage_competitions")
```

Comment:
This kind of code looks like it would be common to many API calls, and if so I think it's important to keep it in a dry place. If that code is called from a `before_action` or from each controller method is less important.